sql/parse/parse.go

```diff
@@ -3179,6 +3179,31 @@ func ExprToExpression(ctx *sql.Context, e sqlparser.Expr) (sql.Expression, error
 	}
 }
 
+// handleCollateExpr is meant to handle generic text-returning expressions that should be reinterpreted as a different collation.
+func handleCollateExpr(ctx *sql.Context, charSet sql.CharacterSetID, expr *sqlparser.CollateExpr) (sql.Expression, error) {
+	//TODO: rename this from Charset to Collation
+	collation, err := sql.ParseCollation(nil, &expr.Charset, false)
+	if err != nil {
+		return nil, err
+	}
+	if collation.Sorter() == nil {
+		return nil, sql.ErrUnsupportedFeature.New("unsupported collation: " + collation.Name())
+	}
+	if collation.CharacterSet() != charSet {
+		//TODO: actual error
+		return nil, fmt.Errorf("COLLATION '%s' is not valid for CHARACTER SET '%s'", collation.Name(), charSet.Name())
+	}
+	innerExpr, err := ExprToExpression(ctx, expr.Expr)
+	if err != nil {
+		return nil, err
+	}
+	if !sql.IsText(innerExpr.Type()) {
```

Comment:
Can't check this here -- it could be a column, which won't have a type until after analysis